### ISSUE TITLE
Merge deploy-appsync and update-appsync

### ISSUE THREAD

**Issue by patrick91** [sid88in/serverless-appsync-plugin#76]
I was thinking that maybe the deploy and update commands should be only one.
The plugin should be able to get the apiId from the existing (if any) appsync api.

What do you think?

**Comment by honkskillet** [comment]
deploy-appsync would have to write to the apiId and apiKey to the yml, but I think this would be a good idea.  It would more closely mimic how serverless deploy works (with just a deploy).

**Comment by patrick91** [comment]
@honkskillet yes! I haven't tried yet but I guess there must be a way to get the apiId and the apiKey using the api name, I'm quite sure we can expect user to have unique api names

**Comment by honkskillet** [comment]
If you run deploy-appsync twice in a row it will create 2 api with the same name but different apiIds.  Using the AWS SDK we can get a list of the APIs to check against. 
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/AppSync.html#listGraphqlApis-property
 If an api with the given name in the yml already exists, we can abort or overwrite. 
We could probably also use this function (and others in the SDK) to generate a serverless/serverless-appsync-plugin project from an existing appsync API.  This can be handy because the Appsync Console does some useful scaffolding when you first start up a project (ie auto-creating the templates from the graphQL).  You could use the AWS web console just to get started then pull the project down to a serverless project so you could continue coding in your local text editor.

**Comment by patrick91** [comment]
Yup, sound like a good idea!

**Comment by sid88in** [comment]
thanks guys. good to have but low priority

**Comment by sid88in** [comment]
1. codegen https://github.com/sid88in/serverless-appsync-plugin/issues/62
2. complete authentication flow - https://github.com/sid88in/serverless-appsync-plugin/issues/70
3. travis build and prettier - https://github.com/sid88in/serverless-appsync-plugin/issues/66 and https://github.com/sid88in/serverless-appsync-plugin/issues/64
4. offline support https://github.com/sid88in/serverless-appsync-plugin/issues/75

**Comment by honkskillet** [comment]
For Auth were you thinking AWS Cognito?

**Comment by kevin-wolf** [comment]
@sid88in doesn't AppSync support authentication through cognito out-of-the-box?

**Comment by honkskillet** [comment]
My stack is an appSync + serverless + IAM/cognito auth + lambda resolver (for mongoDB).  All the resources end up in one cloud formation (except the mongoDB).  I could probably post my serverless.yml.

**Comment by ajmath** [comment]
I was able to find this gist https://gist.github.com/adrianhall/50e9fdf08e7a7e52d3ab0f01467b72f7

The downside is that managing all of the query/subscription/mutation mappings by hand is going to be prone to error and a configuration hell.  Someone should make a serverless plugin huh 😉.

I created #90 to further track the development of the cloudformation approach.  If that goes forward I believe we can close this.
